clients/include/utility.hpp

```diff
@@ -846,6 +861,49 @@ static inline hipDoubleComplex testing_neg(hipDoubleComplex val)
     return ret;
 }
 
+template <typename T>
+hipsparseStatus_t host_nnz(hipsparseDirection_t      dirA,
+                           int                       m,
+                           int                       n,
+                           const hipsparseMatDescr_t descrA,
+                           const T*                  A,
+                           int                       lda,
+                           int*                      nnzPerRowColumn,
+                           int*                      nnzTotalDevHostPtr)
+{
+    int mn = (dirA == HIPSPARSE_DIRECTION_ROW) ? m : n;
+    for(int j = 0; j < mn; ++j)
+    {
+        nnzPerRowColumn[j] = 0;
+    }
+
+    for(int j = 0; j < n; ++j)
```

Comment:
Not sure if we use openmp in hipsparse but you could here if that makes sense

Reply:
Agree!

Reply:
Added some.